### DISCUSSION TITLE
How to disable preview entirely ?

### DISCUSSION THREAD

**Discussion by ysalmon** [texstudio-org/texstudio#2584] [Q&A]
Hi,

I am not using the preview feature of TXS at all. However, I see from time to time messages like
```
Processus en cours : lualatex -synctex=0 -interaction=nonstopmode "NQWrEA".tex

Processus terminé normalement
```
which indicates that some (useless) compilation was done, and this bugs me. In the configuration panel, I have set "Automatic update" to Never and unchecked everything, but it still happens.

**Comment by sunderme** [comment]
i guess , it is the tooltip setting.
<img width="1163" alt="grafik" src="https://user-images.githubusercontent.com/14033169/190577820-f50d0874-90f1-4b31-821a-b19acaf973d2.png">

**Comment by ysalmon** [comment]
It is unchecked.

**Comment by sunderme** [comment]
then, more context is needed.

**Comment by ysalmon** [comment]
About what ?

**Comment by ysalmon** [comment]
Right now it happened again, after resuming my computer from suspend. Last time was just upon opening TXS (and the file).

**Comment by sunderme** [comment]
the completer offers some preview capability (colors)

**Comment by ysalmon** [comment]
Indeed, it seems that hoovering over a ``\textcolor{blue}`` triggers a compilation. I have just unchecked the option to preview colours in the completer, I will see if the compilations stop.

**Comment by ysalmon** [comment]
Well, they do not. Or do I need to restart TXS ?

**Comment by sunderme** [comment]
there seems to be no switch to turn that specific preview off

**Comment by ysalmon** [comment]
So this discussion would be upgraded to a RFE ?

**Reply by sunderme** [reply]
rfe ?

**Comment by ysalmon** [comment]
Request for Enhancement.

And it's definitely the colours : my ``/tmp`` is littered with 24 ``.tex`` containing my preamble and
```latex
\usepackage[active,tightpage]{preview}
\usepackage{varwidth}
\AtBeginDocument{\begin{preview}\begin{varwidth}{\linewidth}}
\AtEndDocument{\end{varwidth}\end{preview}}
\pagestyle{empty}
\begin{document}
\noindent{\color{black} \rule{1cm}{1cm} }
\end{document}
```

**Comment by octaeder** [comment]
see my issue #2269 
Answer was: Intended behavior.

**Comment by octaeder** [comment]
Your first post shows lualatex running. So you use lualatex for preview (not pfdlatex, s. image above by sunderme), right? There is a known issue when you include package tikz (or others?). Precompiling of the preamble (the part before document environment) may fail. Precompiling makes preview faster, but in this case you would have to disable the option ![grafik](https://user-images.githubusercontent.com/102688820/194917701-8d587627-37c7-410a-81e9-347bf4787148.png) in category Preview.

There is an option in category Completion:
![grafik](https://user-images.githubusercontent.com/102688820/194920045-0963ccd1-0e11-4999-8424-e752d7bd596a.png)
But even after restarting txs (without that option checked) colors in \color{} are previewed.

Interesting, there's this option in category Adv. Editor (needs to check Show Advanced Options) that also addresses images:
![grafik](https://user-images.githubusercontent.com/102688820/194922160-c109a110-567b-4f54-90ae-cce05ef814c0.png)

**Comment by octaeder** [comment]
Note: Disabeling Adv. Option

![grafik](https://user-images.githubusercontent.com/102688820/194922160-c109a110-567b-4f54-90ae-cce05ef814c0.png)

suppresses color preview builds (s. #3195, released with txs 4.6.0).
